### PR TITLE
#14677 Add support for caching prebuilt vsix...

### PR DESCRIPTION
**Pull request by nickboldt**
#14677 Add support for caching prebuilt vsix extensions (from mkuznets / mkuznyetsov)

Signed-off-by: nickboldt <nboldt@redhat.com>

### CODE REVIEW

**Comment by nickboldt** [comment]
this PR dupes https://github.com/redhat-developer/codeready-workspaces/pull/188 (minus the Jenkinsfiles which can be added later when ported to run on eclipse CI or wherever)

File: build/scripts/cache_artifacts.sh [hunk 1]
```diff
@@ -23,12 +23,43 @@ fi
...
+
+    if [[ ${rel_plugin_file_path} == ${extension_location} ]]; then
+      matched_plugin_path=${plugin_file_path}
+      echo "found prebuilt extension: ${matched_plugin_path}    "
+      break
+    fi
+  done
+
+  if [[ ! -z "$matched_plugin_path" ]]; then
+    mv "${matched_plugin_path}" ${TEMP_DIR}
```
Comment (amisevsk) [inline build/scripts/cache_artifacts.sh]: This feels like a somewhat hack-y solution.

**Comment by nickboldt** [comment]
Thanks for review @amisevsk ... @mkuznyetsov some feedback for you and work for Monday morning. :)

**Comment by nickboldt** [comment]
After only 24 days of waiting for input from @mkuznyetsov he approved so I merged Angel's changes to my PR into this PR. @amisevsk please re-review. Maybe we can merge this before another 3 weeks go by??

**Comment by nickboldt** [comment]
@amisevsk @mkuznyetsov so... any chance this will get merged into master once CRW 2 is out the door? I don't like seeing PRs left open and forgotten for months as they end up getting outdated and unmergeable. :(

**Comment by amisevsk** [comment]
@nickboldt I'll take a look soon, was a *little* busy these last few weeks :)

**Comment by nickboldt** [comment]
pfft. busy? what's that? :P

**Comment by nickboldt** [comment]
New approach discussed today with @mkuznyetsov will be that we publish all the 3rd party vsix files onto dl.jb.org so that we can reference them from both the Che and CRW plugin registry builds; for CRW, we'll just cache those at build time in Jenkins for reuse in downstream Brew build.

So creation & injection of a vsix.tar.gz file may in fact not be needed in upstream, and can be removed from downstream once we have everything we need on dl.jb.org (3rd party and RH plugins, with source tarballs) or GH (RH plugins only).

**Pull request closed without merging**